src/main/java/org/folio/circulation/domain/MultipleRecords.java

```diff
@@ -113,4 +113,12 @@ public JsonObject asJson(
   public Integer getTotalRecords() {
     return totalRecords;
   }
+
+  public <R> Result<MultipleRecords<R>> map(Function<T, Result<R>> mapper) {
```

Comment:
Hmm, this isn't really a straight map. The mapping function returns a result, which probably makes it a `flatMap` 

And it is mapping the list elements not the records itself.  I will try to think what a good name for that might be.